### PR TITLE
Add a new default overload

### PR DESCRIPTION
**Pull request by Montago**
Slice(int) automatically cuts until the end of the array.

### CODE REVIEW

File: MoreLinq/Slice.cs [hunk 1]
```diff
@@ -51,6 +51,11 @@ public static IEnumerable<T> Slice<T>(this IEnumerable<T> sequence, int startInd
                 ? SliceImpl(asList, startIndex, count)
                 : SliceImpl(sequence, startIndex, count);
         }
+        
+        public static IEnumerable<T> Slice<T>(this IEnumerable<T> sequence, int startIndex)
```
Comment (fsateler) [inline MoreLinq/Slice.cs]: Missing xmldoc

File: MoreLinq/Slice.cs [hunk 2]
```diff
@@ -51,6 +51,11 @@ public static IEnumerable<T> Slice<T>(this IEnumerable<T> sequence, int startInd
...
+        {
+            return sequence.Slice(startIndex, sequence.Count - startIndex);
```
Comment (fsateler) [inline MoreLinq/Slice.cs]: Missing argument null checks, and `IEnumerable` does not have a `Count` property. Moreover, just invoking `Count()` would iterate the sequence twice.

**Comment by atifaziz** [comment]
This overload doesn't make sense as it's redundant with `Skip` that's already there in LINQ. `Skip` is optimised for lists. `Slice(index, count)` solely exists as an optimisation of `Skip` + `Take` for lists.

**Comment by Montago** [comment]
In that case there's no need for Slice either, since it's just a wrapper for Skip + Take

**Pull request closed without merging**